.github/workflows/publish-image.yml

```diff
@@ -0,0 +1,49 @@
+name: Docker Image Build and Publish
+
+on:
+  push:
+    branches: [ "main" ]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+
+    - name: Checkout repository
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+
+    - name: Extract metadata (tags, labels) for Image
+      id: meta
+      uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81 # v5
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@68827325e0b33c7199eb31dd4e31fbe9023e06e3 # v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3
+
+    - name: Log in to the Container registry
+      uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Not sure what is the intention here. Should we configure those additional repository secrets? We already have secrets to push the images to quay.io. I think that we should use those instead.

Reply:
I'm don't have access to see those secrets so I preferred to use the ghcr.io registry and the `GITHUB_TOKEN` which is automatically created at the start of each workflow job.

Reply:
We had problems in the past publishing images in multiple registries. We officially use quay.io to publish Argo related images now. Images for this repo are published here:
https://quay.io/repository/argoprojlabs/argocd-extension-installer?tab=tags
You can use the variables `QUAY_USERNAME` and `QUAY_TOKEN` for authenticating.

Reply:
done, switched to the quay related envs